File: 392.is-subsequence/solution.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        var p = 0
        var q = 0
        while (p < s.length && q < t.length) {
            if (s[p] == t[q++]) {
                p++
            }
        }
        return p == s.length
    }
}

File: 392.is-subsequence/solution.go
func isSubsequence(s string, t string) bool {
    for i, j := 0, 0; i < len(s); j++ {
        if j == len(t) {
            return false
        }
        if s[i] == t[j] {
            i++
        }
    }
    return true
}